modules/createServerRenderContext.js

```diff
@@ -1,34 +1,36 @@
-const k = () => {}
-
 const createServerRenderContext = () => {
   let flushed = false
+  let flushedIndex = 0
   let redirect = null
   let matchContexts = [
-    /* { hasMissComponent: bool, matchesByIdentity: [] } */
+    /* { hasMissComponent: bool, hasMatches: bool } */
   ]
 
-  const setRedirect = flushed ? k : (location) => {
+  const setRedirect = (location) => {
     if (!redirect)
       redirect = location
   }
 
-  const registerMatchContext = flushed ? k : (matchesByIdentity) => {
-    return matchContexts.push({
-      hasMissComponent: false,
-      matchesByIdentity
-    }) - 1
+  // on the second pass, index is determined by render order
+  const registerMatchProvider = () => {
+    return flushed ? flushedIndex++ : (
```

Comment:
Because `registerMatchContext` is called in `componentWillMount`, and (I think) `componentWillMount` is called in sequential order in the children - it is determinstic.

But also: what is the reason for `flushedIndex`? Doesn't seem to be used anywhere apart from the incrementing.